src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java

```diff
@@ -0,0 +1,430 @@
+    public List<Integer> GVCFGQBands = new ArrayList<>();
+    {
+        GVCFGQBands.add(20); GVCFGQBands.add(100);
+    }
+
+    @Advanced
+    @Argument(fullName="drop-low-quals", shortName="drop-low-quals", doc="Exclude variants and homRef blocks that are GQ0 from the reblocked GVCF to save space; drop low quality/uncalled alleles")
+    protected boolean dropLowQuals = false;
+
+    @Advanced
+    @Argument(fullName="rgq-threshold-to-no-call", shortName="rgq-threshold", doc="Reference genotype quality (PL[0]) value below which variant sites will be converted to GQ0 homRef calls")
+    protected double rgqThreshold = 0.0;
+
+    @Advanced
+    @Argument(fullName="do-qual-score-approximation", shortName="do-qual-approx", doc="Add necessary INFO field annotation to perform QUAL approximation downstream")
+    protected boolean doQualApprox = false;
+
+    /**
+     * The rsIDs from this file are used to populate the ID column of the output.  Also, the DB INFO flag will be set when appropriate. Note that dbSNP is not used in any way for the calculations themselves.
+     */
+    @ArgumentCollection
+    protected DbsnpArgumentCollection dbsnp = new DbsnpArgumentCollection();
+
+    // the genotyping engine
+    private HaplotypeCallerGenotypingEngine genotypingEngine;
+    // the annotation engine
+    private VariantAnnotatorEngine annotationEngine;
+    // the INFO field annotation key names to remove
+    private final List<String> infoFieldAnnotationKeyNamesToRemove = Arrays.asList(GVCFWriter.GVCF_BLOCK,
+            GATKVCFConstants.DOWNSAMPLED_KEY, GATKVCFConstants.HAPLOTYPE_SCORE_KEY,
+            GATKVCFConstants.INBREEDING_COEFFICIENT_KEY, GATKVCFConstants.MLE_ALLELE_COUNT_KEY,
+            GATKVCFConstants.MLE_ALLELE_FREQUENCY_KEY);
+
+    private VariantContextWriter vcfWriter;
+
+    @Override
+    public boolean useVariantAnnotations() { return true;}
+
+    @Override
+    public List<Annotation> getDefaultVariantAnnotations() {
+        return Arrays.asList(new Coverage(), new RMSMappingQuality(), new ReadPosRankSumTest(), new MappingQualityRankSumTest());
+
+    }
+
+    public void onTraversalStart() {
+        VCFHeader inputHeader = getHeaderForVariants();
+        final Set<VCFHeaderLine> inputHeaders = inputHeader.getMetaDataInSortedOrder();
+
+        final Set<VCFHeaderLine> headerLines = new HashSet<>(inputHeaders);
+        // Remove GCVFBlocks, legacy headers, and annotations that aren't informative for single samples
+        headerLines.removeIf(vcfHeaderLine -> vcfHeaderLine.getKey().startsWith(GVCFWriter.GVCF_BLOCK) ||
+                (vcfHeaderLine.getKey().equals("INFO")) && infoFieldAnnotationKeyNamesToRemove.contains(((VCFInfoHeaderLine)vcfHeaderLine).getID()));
+
+        headerLines.addAll(getDefaultToolVCFHeaderLines());
+
+        genotypingEngine = createGenotypingEngine(new IndexedSampleList(inputHeader.getGenotypeSamples()));
```

Comment:
What is the behavior if the input has multiple samples ? Should this verify that there is only a single sample in the input ?

Reply:
This seems like a good check to add.

Reply:
Oh, you did add it up above 👍